server/src/main/java/io/crate/replication/logical/plan/DropSubscriptionPlan.java

```diff
@@ -62,59 +52,10 @@ public void executeOrFail(DependencyCarrier dependencies,
-
-        final List<ChainableAction<Long>> actions = new ArrayList<>();
-
-        // Step 1 - Close subscribed tables and consequently stop tracking and remove retention lease.
-        actions.add(new ChainableAction<>(
-            () -> dependencies.alterTableOperation()
-                .executeAlterTableOpenClose(tables, false, true, null)
-                .exceptionally(error -> {
-                    throw new IllegalStateException(
-                        "Couldn't close subscribed tables, please retry DROP SUBSCRIPTION command."
-                            + error.getMessage(), error
-                    );
-                }),
-            () -> CompletableFuture.completedFuture(-1L)
-        ));
-
-        // Step 2
-        // Drop setting and subscription
-        actions.add(new ChainableAction<>(
-            () -> {
-                FutureActionListener<AcknowledgedResponse, Long> listener = new FutureActionListener<>(r -> 1L);
-                var request = new DropSubscriptionRequest(analyzedDropSubscription.name(), analyzedDropSubscription.ifExists());
-                dependencies.dropSubscriptionAction().execute(request, listener);
-                return listener.exceptionally(error -> {
-                    throw new IllegalStateException(
-                        "Couldn't update metadata, please retry DROP SUBSCRIPTION command."
-                            + error.getMessage(), error
-                    );
-                });
-            },
-            () -> CompletableFuture.completedFuture(-1L)
-        ));
-
-        // Step 3
-        // Reopen table to update table engine to normal (Operation will be reset back to ALL)
-        actions.add(new ChainableAction<>(
-            () -> dependencies.alterTableOperation()
-                .executeAlterTableOpenClose(tables, true, true, null)
-                .exceptionally(error -> {
-                    String tablesHint = "all subscribed tables. ";
-                    if (tables.size() < 5) {
-                        tablesHint = "tables (" + tables.stream().map(relationName -> relationName.name()).collect(Collectors.joining(",")) + "). ";
-                    }
-                    throw new IllegalStateException(
-                        "Couldn't reopen tables, please run command ALTER TABLE OPEN for " + tablesHint
-                            + error.getMessage(), error
-                    );
-                }),
-            () -> CompletableFuture.completedFuture(-1L)
-        ));
-
-        ChainableActions.run(actions).whenComplete(new OneRowActionListener<>(consumer, rCount -> new Row1(rCount == null ? -1 : rCount)));
+        var request = new DropSubscriptionRequest(analyzedDropSubscription.name(), analyzedDropSubscription.ifExists());
+        dependencies.dropSubscriptionAction().execute(
+            request,
+            new OneRowActionListener<>(consumer, rCount -> new Row1(rCount == null ? -1 : 1L))
+        );
```

Comment:
Maybe already use the client given that we intend to remove the transport actions from the dependencies.

But also fine to do this as follow up.